File: src/main/java/com/example/ticketRush/BookingModule/Service/BookingService.java
package com.example.ticketRush.BookingModule.Service;

import com.example.ticketRush.BookingModule.Dto.Request.BookingHoldRequest;
import com.example.ticketRush.BookingModule.Dto.Response.BookingResponse;
import org.springframework.security.core.Authentication;

import java.util.List;
import java.util.UUID;

public interface BookingService {

    BookingResponse holdSeats(Authentication authentication, BookingHoldRequest request);

    BookingResponse getBookingById(Authentication authentication, UUID bookingId);

    BookingResponse confirmBooking(Authentication authentication, UUID bookingId);

    List<BookingResponse> getMyBookings(Authentication authentication);
}

File: FE/src/app/services/booking-service.ts
import apiClient from "../api-client";
import { Booking } from "../types";

export interface BookingHoldRequest {
  eventId: number;
  seatIds: number[];
}

export const bookingService = {
  holdSeats: async (request: BookingHoldRequest): Promise<Booking> => {
    const response = await apiClient.post<Booking>("/api/bookings/hold", request);
    return response.data;
  },

  getBookingById: async (bookingId: string): Promise<Booking> => {
    const response = await apiClient.get<Booking>(`/api/bookings/${bookingId}`);
    return response.data;
  },

  confirmBooking: async (bookingId: string): Promise<Booking> => {
    const response = await apiClient.post<Booking>(`/api/bookings/${bookingId}/confirm`);
    return response.data;
  },

  getMyBookings: async (): Promise<Booking[]> => {
    const response = await apiClient.get<Booking[]>("/api/bookings/me");
    return response.data;
  },
};
